### PR TITLE
@skip, @include directives for objects

### PR DESCRIPTION
**Pull request by gilgardosh**
Related #5238

### CODE REVIEW

File: packages/plugins/flow/operations/tests/flow-documents.spec.ts [hunk 1]
```diff
@@ -1251,9 +1257,13 @@ describe('Flow Operations Plugin', () => {
       expect(result).toBeSimilarStringTo(`
       export type UserQueryVariables = {
         showAddress: $ElementType<Scalars, 'Boolean'>,
+        showName: $ElementType<Scalars, 'Boolean'>,
       };
 
-      export type UserQuery = { user: $MakeOptional<$Pick<User, { id: *, name?: *, address: * }>, { address: * }> };
+      export type UserQuery = { user: ({
```
Comment (gilgardosh) [inline packages/plugins/flow/operations/tests/flow-documents.spec.ts:1263]: @dotansimha Please make sure that's the wanted outcome

**Comment by dotansimha** [comment]
Thanks @gilgardosh , what about `preResolveTypes`? can you please add a test for it?

**Comment by dotansimha** [comment]
Thanks @gilgardosh . as discussed offline - please clean it a bit and then we can merge :)

**Comment by dotansimha** [comment]
Awesome work. thank you @gilgardosh !

File: packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts [hunk 2]
```diff
@@ -372,14 +372,15 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
...
 
       linkFields.push({
         alias: field.alias ? this._processor.config.formatNamedField(field.alias.value, selectedFieldType) : undefined,
-        name: this._processor.config.formatNamedField(field.name.value, selectedFieldType),
+        name: this._processor.config.formatNamedField(field.name.value, selectedFieldType, isConditional),
         type: realSelectedFieldType.name,
         selectionSet: this._processor.config.wrapTypeWithModifiers(
           selectionSet.transformSelectionSet().split(`\n`).join(`\n  `),
-          selectedFieldType
+          isConditional ? realSelectedFieldType : selectedFieldType
```
Comment (raysuelzer) [inline packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts:383]: Breaks arrays as this strips the interable wrapper (Array<T>) from the base type when in a directive. 

EG if people is a list: 
```
people @include(if: $withPeople) { 
 id
}
```
Result is that in typescript the output for People is no longer Maybe<Array<Person>> but just Maybe<Person>

**Pull request merged**